contrib/epee/src/net_helper.cpp

```diff
@@ -52,12 +23,12 @@ namespace net_utils
 			boost::promise<boost::asio::ip::tcp::socket> result_;
 			boost::asio::ip::tcp::socket socket_;
 
-			explicit new_connection(boost::asio::io_service& io_service)
+			explicit new_connection(const boost::asio::steady_timer::executor_type& io_service)
```

Comment:
This should fail on the `socket_` constructor on Boost 1.66-1.69. The problem is that `socket_` will require an `io_context` object, whereas newer Boost versions only need an `executor`. This will likely be the source of many compile errors moving forward.

I'm not certain how I will fix this, I may have to change this function signature which may break some downstream projects.

Reply:
Fixed.